compiler/rustc_mir/src/transform/instcombine.rs

```diff
@@ -296,3 +298,12 @@ struct OptimizationList<'tcx> {
     unneeded_equality_comparison: FxHashMap<Location, Operand<'tcx>>,
     unneeded_deref: FxHashMap<Location, Place<'tcx>>,
 }
+
+impl<'tcx> OptimizationList<'tcx> {
+    fn is_empty(&self) -> bool {
+        self.and_stars.is_empty()
+            && self.arrays_lengths.is_empty()
+            && self.unneeded_equality_comparison.is_empty()
+            && self.unneeded_deref.is_empty()
```

Comment:
I think this should probably use an exhaustive pattern match so that it is updated if more fields are added:

```suggestion
        let OptimizationList { and_stars, arrays_length, unneeded_equality_comparison, unneeded_deref } = self;
        and_stars.is_empty()
            && arrays_lengths.is_empty()
            && unneeded_equality_comparison.is_empty()
            && unneeded_deref.is_empty()
```

(You'll probably need to run `./x.py fmt` though)